addons/calendar/models/calendar.py

```diff
@@ -0,0 +1,1092 @@
+                access_token = self.new_invitation_token(event, partner.id)
+                values = {
+                    'partner_id': partner.id,
+                    'event_id': event.id,
+                    'access_token': access_token,
+                    'email': partner.email,
+                }
+
+                if partner.id == self.env.user.partner_id.id:
+                    values['state'] = 'accepted'
+
+                attendee = self.env['calendar.attendee'].create(values)
+                new_attendees.append(attendee)
+                new_att_partner_ids.append(partner.id)
+
+                if not self.env.user.email or self.env.user.email != partner.email:
+                    mail_from = self.env.user.email or tools.config.get('email_from', False)
+                    if not self._context.get('no_email'):
+                        attendee._send_mail_to_attendees(email_from=mail_from)
+
+            if self.env.user.partner_id.id not in new_att_partner_ids:
+                access_token = self.new_invitation_token(event, self.env.user.partner_id.id)
+                values = {
+                    'partner_id': self.env.user.partner_id.id,
+                    'event_id': event.id,
+                    'access_token': access_token,
+                    'email': self.env.user.partner_id.email,
+                }
+                current_user_att_id = self.env['calendar.attendee'].create(values)
+                new_attendees.append(current_user_att_id)
+                new_att_partner_ids.append(self.env.user.partner_id.id)
+            if new_attendees:
+                event.write({'attendee_ids': [(4, att.id) for att in new_attendees]})
+            if new_att_partner_ids:
+                event.message_subscribe(new_att_partner_ids)
+
+            # We remove old attendees who are not in partner_ids now.
+            all_partner_ids = [part.id for part in event.partner_ids]
+            all_part_attendee_ids = [att.partner_id.id for att in event.attendee_ids]
+            all_attendee_ids = [att.id for att in event.attendee_ids]
+            partner_ids_to_remove = map(lambda x: x, set(all_part_attendee_ids + new_att_partner_ids) - set(all_partner_ids))
+            attendee_ids_to_remove = []
+
+            if partner_ids_to_remove:
+                attendee_ids_to_remove = self.env["calendar.attendee"].search([('partner_id.id', 'in', partner_ids_to_remove), ('event_id.id', '=', event.id)])
+                if attendee_ids_to_remove:
+                    attendee_ids_to_remove.unlink()
+
+            res[event.id] = {
+                'new_attendee_ids': new_attendees,
+                'old_attendee_ids': all_attendee_ids,
+                'removed_attendee_ids': attendee_ids_to_remove
+            }
+        return res
+
+    @api.multi
+    def get_search_fields(self, order_fields, r_date=None):
+        sort_fields = {}
+        for ord in order_fields:
+            if ord == 'id' and r_date:
```

Comment:
ord is python builtin function, Return the integer ordinal of a one-character string.